vm_eval.c

```diff
@@ -2169,35 +2176,55 @@ specific_eval(int argc, const VALUE *argv, VALUE self, int singleton, int kw_spl
 
 /*
  *  call-seq:
- *     obj.instance_eval(string [, filename [, lineno]] )   -> obj
- *     obj.instance_eval {|obj| block }                     -> obj
+ *    instance_eval {|self| ... } -> object
```

Comment:
`self` is a keyword so it can't actually use it as a variable name:

```ruby
1.times { |self| }
# SyntaxError: syntax error, unexpected `self', expecting '|'
```

Can we go back to using `obj`?